### PR TITLE
Abhi/collect metrics

### PR DESCRIPTION
**Pull request by pretentious7**

### CODE REVIEW

File: plane2/src/drone/backend_manager.rs [hunk 1]
```diff
@@ -222,6 +251,31 @@ impl BackendManager {
...
+        handle.take();
+
+        let docker = self.docker.clone();
+        let id = self.container_id.clone();
+        let self_clone = self.clone();
+
+        *handle = Some(GuardHandle::new(async move {
+            while {
+                tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+                true
```
Comment (paulgb) [inline plane2/src/drone/backend_manager.rs:270]: using [interval](https://docs.rs/tokio/latest/tokio/time/fn.interval.html) should give slightly more uniform time-steps

File: plane2/src/drone/backend_manager.rs [hunk 2]
```diff
@@ -38,17 +41,36 @@ impl StepStatusResult {
     }
 }
 
-type StateCallback = Box<dyn Fn(&BackendState) -> Result<(), Box<dyn Error>> + Send + Sync>;
+struct Callback<T>(Box<dyn Fn(T) -> Result<(), Box<dyn Error>> + Send + Sync>);
```
Comment (paulgb) [inline plane2/src/drone/backend_manager.rs:44]: Is there an advantage to using a newtype here over a type alias? Is it that you wanted Debug implemented?

File: plane2/src/drone/backend_manager.rs [hunk 3]
```diff
@@ -59,7 +81,10 @@ pub struct BackendManager {
     executor_config: ExecutorConfig,
 
     /// Function to call when the state changes.
-    state_callback: StateCallback,
+    state_callback: Callback<BackendState>,
+
+    /// Function to call when metrics produced.
+    metrics_callback: Callback<Stats>,
```
Comment (paulgb) [inline plane2/src/drone/backend_manager.rs:87]: Given that this callback will likely just pass the data over the socket, should we just use a `TypedSender` for this? This PR gave me an idea for improving the ergonomics of `TypedSender` for this purpose, which I opened here: https://github.com/drifting-in-space/plane/pull/508

File: plane2/src/drone/backend_manager.rs (hunk 2 above)
Comment (pretentious7) [reply plane2/src/drone/backend_manager.rs:44]: So i can stick tracing::instrument on things to debug.

**Comment by pretentious7** [comment]
superseded by https://github.com/drifting-in-space/plane/pull/539

**Pull request closed without merging**